R/PseudoBulk.R

```diff
@@ -526,14 +526,18 @@ RunFilteredContrasts <- function(seuratObj, filteredContrastsFile = NULL, filter
   }
   
   results <- future.apply::future_lapply(split(contrasts, 1:nrow(contrasts)), future.seed = GetSeed(), FUN = function(x){
-    #initialize two seurat objects, one to be subset according to the positive contrasts, and one to be subset according to the negative contrast. These will be merged downstream.
+    #initialize two Seurat objects, one to be subset according to the positive contrasts, and one to be subset according to the negative contrast. These will be merged downstream.
     seuratObj.positive.contrast <- seuratObj
     seuratObj.negative.contrast <- seuratObj
     positive_contrast <- NULL
     negative_contrast <- NULL
     #construct the positive and negative contrast by accessing each value within the filtered contrast data frame (row-wise) by adding the appropriate prefix to the metadata field name (i.e. the "contrast column")
     print(paste0("Contrast columns: ", attr(design,"contrast_columns")))
     for (contrast_column in attr(design, "contrast_columns")){
+      #check if the contrast column in the parent Seurat object needs sanitizing before populating seuratObj.positive.contrast and seuratObj.negative.contrast downstream.
+      if (!all(seuratObj@meta.data[,contrast_column] == make.names(seuratObj@meta.data[,contrast_column]))){
+        seuratObj@meta.data[,contrast_column] <- make.names(seuratObj@meta.data[,contrast_column])
```

Comment:
could you at least log a line to alert the user this is happening? this would also convert factors into non-factors, right?

would it make sense to test if it is a factor, and if true, should we do someone using the forcats package to more surgically rename factor levels? this would preserve the order, if that was deliberate.